docfx_yaml/extension.py

```diff
@@ -1348,6 +1363,61 @@ def pretty_package_name(package_group):
     return " ".join(capitalized_name)
 
 
+# Type alias used for yaml entries.
+_yaml_type_alias = dict[str, any]
+
+
+def _find_summary_details(
+    yaml_data: _yaml_type_alias,
+    summary_type: str,
+    cgc_url: str,
+) -> _yaml_type_alias:
+    """Finds the summary details to add for a given entry."""
+    uid = yaml_data.get("uid", "")
+    item_to_add = uid if summary_type == CLASS else f"{uid}-summary"
+
+    if item_to_add not in _ENTRY_NAME_AND_ENTRY_CONTENT_BY_SUMMARY_TYPE[summary_type][0]:
+        _ENTRY_NAME_AND_ENTRY_CONTENT_BY_SUMMARY_TYPE[summary_type][0].append(
+            item_to_add
+        )
+
+    if summary_type in [CLASS]:
+        name_to_use = f"[{uid}]({cgc_url}{uid})"
+
+    return {
+        "uid": uid,
+        "name": name_to_use,
+        "fullName": uid,
+        "isExternal": False,
+    }
+
+
+def _render_summary_content(
+    children_name_and_summary_content: Sequence[Sequence[str]],
+    entry_name: str,
+    summary_type: str,
+    library_name: str,
+) -> _yaml_type_alias:
+    """Returns the summary content in appropriate YAML format to write."""
+    summary_content = {}
+
+    if summary_type in [CLASS]:
+        summary_content = {
+            "items": [{
+                'uid': f'{summary_type.lower()}-summary',
+                'name': entry_name,
+                'fullName': f'{entry_name} Summary',
+                'langs': ['python'],
+                'type': 'package',
+                'summary': f'Summary of entries of {entry_name} for {library_name}.',
+                'children': children_name_and_summary_content[0],
```

Comment:
Doesn't this need to be converted into a YAML list?
Something like:
```
reduce(lambda x, y:  y + '\n - '  + x , children_name_and_summary_content[0])
```

Reply:
Nope, that gets handled already!